datasets/subjqa/subjqa.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{bjerva20subjqa,
+    title = "SubjQA: A Dataset for Subjectivity and Review Comprehension",
+    author = "Bjerva, Johannes  and
+      Bhutani, Nikita  and
+      Golahn, Behzad  and
+      Tan, Wang-Chiew  and
+      Augenstein, Isabelle",
+    booktitle = "Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing",
+    month = November,
+    year = "2020",
+    publisher = "Association for Computational Linguistics",
+}
+"""
+
+_DESCRIPTION = """SubjQA is a question answering dataset that focuses on subjective questions and answers.
+The dataset consists of roughly 10,000 questions over reviews from 6 different domains: books, movies, grocery,
+electronics, TripAdvisor (i.e. hotels), and restaurants."""
+
+_HOMEPAGE = ""
+
+_LICENSE = ""
+
+_URLs = {"default": "https://github.com/lewtun/SubjQA/archive/refs/heads/master.zip"}
+
+
+class Subjqa(datasets.GeneratorBasedBuilder):
+    """SubjQA is a question answering dataset that focuses on subjective questions and answers."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="books", version=VERSION, description="Amazon book reviews"),
+        datasets.BuilderConfig(name="electronics", version=VERSION, description="Amazon electronics reviews"),
+        datasets.BuilderConfig(name="grocery", version=VERSION, description="Amazon grocery reviews"),
+        datasets.BuilderConfig(name="movies", version=VERSION, description="Amazon movie reviews"),
+        datasets.BuilderConfig(name="restaurants", version=VERSION, description="Yelp restaurant reviews"),
+        datasets.BuilderConfig(name="tripadvisor", version=VERSION, description="TripAdvisor hotel reviews"),
+    ]
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "item_id": datasets.Value("string"),
+                "domain": datasets.Value("string"),
+                "nn_mod": datasets.Value("string"),
+                "nn_asp": datasets.Value("string"),
+                "query_mod": datasets.Value("string"),
+                "query_asp": datasets.Value("string"),
+                "q_review_id": datasets.Value("string"),
+                "q_reviews_id": datasets.Value("string"),
+                "question": datasets.Value("string"),
+                "question_subj_level": datasets.Value("int64"),
+                "ques_subj_score": datasets.Value("float"),
+                "is_ques_subjective": datasets.Value("bool"),
+                "review_id": datasets.Value("string"),
+                "review": datasets.Value("string"),
+                "human_ans_spans": datasets.Value("string"),
+                "human_ans_indices": datasets.Value("string"),
+                "answer_subj_level": datasets.Value("int64"),
+                "ans_subj_score": datasets.Value("float"),
+                "is_ans_subjective": datasets.Value("bool"),
+            }
```

Comment:
Do you think we could have the same feature types as squad ? This way people won't have to transform/rename columns if they take a generic training script that was made for squad to train a model on this dataset.

The squad features are
```python
{
    "id": datasets.Value("string"),
    "title": datasets.Value("string"),
    "context": datasets.Value("string"),
    "question": datasets.Value("string"),
    "answers": datasets.features.Sequence(
        {
            "text": datasets.Value("string"),
            "answer_start": datasets.Value("int32"),
        }
    ),
}
```

Reply:
good idea! 

since squad-formatted datasets group together the answers associated with a question-context pair, e.g.

```json
{
    "answers": {
        "answer_start": [17, 7, 17],
        "text": ["1991", "autumn of 1991", "1991"],
    },
    "context": 'In the autumn of 1991, talks were held for the broadcast rights for Premier League for a five-year period, from the 1992 season. ITV were the current rights holders, and fought hard to retain the new rights. ITV had increased its offer from £18m to £34m per year to keep control of the rights. BSkyB joined forces with the BBC to make a counter bid. The BBC was given the highlights of most of the matches, while BSkyB paying £304m for the Premier League rights, would give them a monopoly of all live matches, up to 60 per year from the 1992 season.  Murdoch described sport as a "battering ram" for pay-television, providing a strong customer base. A few weeks after the deal, ITV went to the High Court to get an injunction as it believed their bid details had been leaked before the decision was taken. ITV also asked the Office of Fair Trading to investigate since it believed Rupert Murdoch\'s media empire via its newspapers had influenced the deal. A few days later neither action took effect, ITV believed BSkyB was telephoned and informed of its £262m bid, and Premier League advised BSkyB to increase its counter bid.',
    "id": "57096e1ced30961900e84102",
    "question": "When were the talks held for braodcast right to the Primier league for a five year period from the 1992 season?",
    "title": "Sky_(United_Kingdom)",
}
```

my proposal is to augment the `answers` field with additional metadata like `is_ans_subjective`, and add the remaining metadata as additional fields alongside `id`, `context`, etc. i'll push the change later today!

Reply:
i've now pushed the changes to create a squad-formatted version of the dataset. 

to keep the logic similar with other squad datasets in `datasets`, i opted to first convert the csv files into squad-format, and then loop over the examples in the same way that's done for squad, squad_v2 etc.